src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -0,0 +1,1159 @@
+
+static void
+call_service_monitor_callback(struct sol_connman_service *service)
+{
+    struct sol_monitors_entry *m;
+    uint16_t i;
+
+    SOL_MONITORS_WALK (&_ctx.service_ms, m, i)
+        ((sol_connman_service_monitor_cb)m->cb)((void *)m->data, service);
+}
+
+static void
+call_manager_monitor_callback(void)
+{
+    struct sol_monitors_entry *m;
+    uint16_t i;
+
+    SOL_MONITORS_WALK (&_ctx.manager_ms, m, i)
+        ((sol_connman_manager_monitor_cb)m->cb)((void *)m->data);
+}
+
+static void
+call_error_monitor_callback(struct sol_connman_service *service,
+    enum sol_connman_error error)
+{
+    struct sol_monitors_entry *m;
+    uint16_t i;
+
+    SOL_MONITORS_WALK (&_ctx.error_ms, m, i)
+        ((sol_connman_error_monitor_cb)m->cb)((void *)m->data,
+            service, error);
+}
+
+static void
+_set_error_to_callback(struct sol_connman_service *service,
+    sd_bus_error *ret_error)
+{
+    enum sol_connman_error error;
+    static const struct sol_str_table err_table[] = {
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.NoMemory",
+            SOL_CONNMAN_ERROR_NO_MEM),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.AccessDenied",
+            SOL_CONNMAN_ERROR_ACCESS_DENIED),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.InvalidArgs",
+            SOL_CONNMAN_ERROR_INVALID_ARGS),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.UnixProcessIdUnknown",
+            SOL_CONNMAN_ERROR_UNIX_PROCESS_ID_UNKNOWN),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.FileNotFound",
+            SOL_CONNMAN_ERROR_FILE_NOT_FOUND),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.FileExists",
+            SOL_CONNMAN_ERROR_FILE_EXISTS),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.Timeout",
+            SOL_CONNMAN_ERROR_TIMEOUT),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.IOError",
+            SOL_CONNMAN_ERROR_IO_ERROR),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.Disconnected",
+            SOL_CONNMAN_ERROR_DISCONNECTED),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.NotSupported",
+            SOL_CONNMAN_ERROR_NOT_SUPPORTED),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.BadAddress",
+            SOL_CONNMAN_ERROR_BAD_ADDRESS),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.LimitsExceeded",
+            SOL_CONNMAN_ERROR_LIMITS_EXCEEDED),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.AddressInUse",
+            SOL_CONNMAN_ERROR_ADDRESS_IN_USE),
+        SOL_STR_TABLE_ITEM("org.freedesktop.DBus.Error.InconsistentMessage",
+            SOL_CONNMAN_ERROR_INCONSISTENT_MESSAGE),
+        { }
+    };
+
+    if (ret_error) {
+        error = sol_str_table_lookup_fallback(err_table,
+            sol_str_slice_from_str(ret_error->name), SOL_CONNMAN_ERROR_UNKNOWN);
+        call_error_monitor_callback(service, error);
+    } else
+        call_error_monitor_callback(service, SOL_CONNMAN_ERROR_NONE);
```

Comment:
I guess no point in call with no-error... 

if we passed it a "async handle", then we could use to inform a specific call succeed, but it's not the case.